components/psst/browser/content/psst_tab_web_contents_observer_unittest.cc

```diff
@@ -225,9 +234,126 @@ TEST_F(PsstTabWebContentsObserverUnitTest, ShouldOnlyProcessHttpOrHttps) {
   }
 }
 
-// TODO(bridiver) - test multiple main frame navigations in a row
-// TODO(bridiver) - test redirects
-// TODO(bridiver) - test for navigating after OnLoad but before script
+TEST_F(PsstTabWebContentsObserverUnitTest,
+       ShouldProcessMultipleMainFrameNavigations) {
+  const std::string first_nav_user_script = "user1";
+  const std::string second_nav_user_script = "user2";
+  const std::string policy_script = "policy";
+  const GURL first_navigation_url("https://example1.com");
+  const GURL second_navigation_url("https://example2.com");
+
+  base::RunLoop first_nav_check_loop;
```

Comment:
Putting everything together at the beginning makes this test a bit confusing to follow. Can you please first define/run everything for the first navigation including run loop waits, etc... and then define/run everything for the second one after that?